server/src/main/java/org/opensearch/cluster/routing/allocation/AllocationService.java

```diff
@@ -548,6 +564,22 @@ private void allocateExistingUnassignedShards(RoutingAllocation allocation) {
             existingShardsAllocator.beforeAllocation(allocation);
         }
 
+        Boolean batchModeEnabled = EXISTING_SHARDS_ALLOCATOR_BATCH_MODE.get(settings);
+
+        if (batchModeEnabled && allocation.nodes().getMinNodeVersion().onOrAfter(Version.CURRENT)) {
+            // since allocators is per index setting, to have batch assignment verify allocators same for all shards
+            // if not fallback to single assignment
+            ExistingShardsAllocator allocator = verifySameAllocatorForAllUnassignedShards(allocation);
+            if (allocator != null) {
+                allocator.allocateUnassignedBatch(allocation, true);
+                for (final ExistingShardsAllocator existingShardsAllocator : existingShardsAllocators.values()) {
+                    existingShardsAllocator.afterPrimariesBeforeReplicas(allocation);
+                }
+                allocator.allocateUnassignedBatch(allocation, false);
+                return;
+            }
```

Comment:
If allocator was found null, then execution would come out of this if block and continue with single shard allocation. Is this correct? Shouldn't we be throwing an exception from `verifySameAllocatorForAllUnassignedShards` and remove this null check altogether?

Reply:
not it wont assign a single shard, but try to assign All unassigned Shards 
@amkhar I think you made some suggestion to rename ```allocateUnassignedBatch``` ```allocateAllUnassignedShards```?

Reply:
Yes, this method ultimately allocates all unassigned shards. And `UnassignedBatch` feels like a single batch is getting assigned. 

So, we can change it to `allocateAllUnassignedShards`. Feel free to suggest other better/relevant names.

Reply:
I don't like this complication. If the cluster is using single ExistingShardAllocator, then use batch mode otherwise let it fall back to single shard allocation. This should change later to batch mode code which is dealing with single ShardRouting once there is more confidence in the new code with batch mode.

Reply:
Add a warning log that it can't use batch due to multiple ExistingShardAllocator, it is recommending to use one so that batch logic can run effectively.